core/server/middleware/error-handler.js

```diff
@@ -103,15 +71,68 @@ module.exports = function errorHandler(err, req, res, next) {
         'Cache-Control': 'no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0'
     });
 
-    // @TODO: does this resolves all use cases?
-    if (!req.headers.accept && !req.headers.hasOwnProperty('content-type')) {
-        req.headers.accept = 'text/html';
+    return err;
+};
+
+errorHandler.resourceNotFound = function resourceNotFound(req, res, next) {
+    // TODO, handle unknown resources & methods differently, so that we can also produce
+    // 405 Method Not Allowed
+    next(new errors.NotFoundError({message: i18n.t('errors.errors.resourceNotFound')}));
+};
+
+errorHandler.pageNotFound = function pageNotFound(req, res, next) {
+    next(new errors.NotFoundError({message: i18n.t('errors.errors.pageNotFound')}));
+};
+
+errorHandler.handleJSONResponse = function _handleJSONResponse(err, req, res, next) {
+    /*jshint unused:false */
+    // Make sure the error can be served
+    err = _private.prepareError(err, req, res);
```

Comment:
suggestion:
Not a big fan of the function design here, that's why i would like to suggest this here:

```
_private.beforeResponse= function(err, req,res) {
   req.set..
   res....
};

_private.prepareError = function (done) {
    return function prepareError(err, req, res, next) {
          // ...
          done(err, req, res,next);
    });
};

errorHandler.handleJSONResponse = _private.prepareError(function _handleJSONResponse(err, req, res, next){
    _private.beforeResponse(err, req, res);
});

```